src/anemoi/models/preprocessing/remapper.py

```diff
@@ -16,315 +16,58 @@
-
-    def inverse_transform(self, x: torch.Tensor, in_place: bool = True) -> torch.Tensor:
-        """Convert and remap the output tensor.
-
-        ```
-        x : torch.Tensor
-            Input tensor
-        in_place : bool
-            Whether to process the tensor in place.
-            in_place is not possible for this postprocessor.
-        ```
-        """
-        # Choose correct index based on number of variables
-        if x.shape[-1] == self.num_remapped_training_output_vars:
-            index = self.index_training_output
-            indices_remapped = self.index_training_backmapped_output
-            indices_keep = self.indices_keep_training_output
-            target_number_columns = self.num_training_output_vars
-
-        elif x.shape[-1] == self.num_remapped_inference_output_vars:
-            index = self.index_inference_output
-            indices_remapped = self.index_inference_backmapped_output
-            indices_keep = self.indices_keep_inference_output
-            target_number_columns = self.num_inference_output_vars
-
-        else:
-            raise ValueError(
-                f"Input tensor ({x.shape[-1]}) does not match the training "
-                f"({self.num_remapped_training_output_vars}) or inference shape ({self.num_remapped_inference_output_vars})",
-            )
-
-        # create new tensor with target number of columns
-        x_remapped = torch.zeros(x.shape[:-1] + (target_number_columns,), dtype=x.dtype, device=x.device)
-        if in_place and not self.printed_postprocessor_warning:
-            LOGGER.warning(
-                "Remapper (preprocessor) called with in_place=True. This preprocessor cannot be applied in_place as new columns are added to the tensors.",
-            )
-            self.printed_postprocessor_warning = True
-
-        # copy variables that are not remapped
-        x_remapped[..., indices_keep] = x[..., : len(indices_keep)]
-
-        # Backmap variables
-        for idx_dst, backmapper, idx_src in zip(index, self.backmappers, indices_remapped):
-            if idx_dst is not None:
-                x_remapped[..., idx_dst] = backmapper(x[..., idx_src])
-
-        return x_remapped
-
-
-class Remapper(BaseRemapperVariable):
-    """Remap and convert variables.
-
-    cos_sin:
-        Remap the variable to cosine and sine.
-        Map output back to degrees.
-
-    ```
-    cos_sin:
-      "mwd" : ["cos_mwd", "sin_mwd"]
-    ```
-    """
-
-    def __init__(
-        self,
-        config=None,
-        data_indices: Optional[IndexCollection] = None,
-        statistics: Optional[dict] = None,
-    ) -> None:
-        super().__init__(config, data_indices, statistics)
-
-        self.printed_preprocessor_warning, self.printed_postprocessor_warning = False, False
-
-        self._create_remapping_indices(statistics)
-
-        self._validate_indices()
+                raise ValueError(f"Unknown remapping method for {name}: {method}")
```

Comment:
Can we follow the style of the other preprocessors for these checks, please, and put them into a` _validate_monomapper_inputs` function, as we do it for example for the normalizer?